extensions/ql-vscode/test/vscode-tests/cli-integration/variant-analysis/variant-analysis-manager.test.ts

```diff
@@ -202,5 +204,136 @@ describe("Variant Analysis Manager", () => {
 
       await expect(promise).rejects.toThrow(UserCancellationException);
     });
+
+    describe("check variant analysis generated packs", () => {
+      beforeEach(() => {
+        mockSubmitVariantAnalysis = jest
+          .spyOn(ghApiClient, "submitVariantAnalysis")
+          .mockResolvedValue({
+            id: 1,
+            query_language: QueryLanguage.Javascript,
+            query_pack_url: "http://example.com",
+            created_at: "2021-01-01T00:00:00Z",
+            updated_at: "2021-01-01T00:00:00Z",
+            status: "in_progress",
+            controller_repo: {
+              id: 1,
+              name: "vscode-codeql",
+              full_name: "github/vscode-codeql",
+              private: false,
+            },
+            actions_workflow_run_id: 20,
+            scanned_repositories: [] as any[],
+          });
+
+        executeCommandSpy = jest.spyOn(commands, "executeCommand");
+      });
+
+      it("should run a remote query that is part of a qlpack", async () => {
+        await doVariantAnalysisTest({
+          queryPath: "data-remote-qlpack/in-pack.ql",
+          filesThatExist: ["in-pack.ql", "lib.qll"],
+          filesThatDoNotExist: [],
+          qlxFilesThatExist: ["in-pack.qlx"],
+        });
+      });
+
+      it("should run a remote query that is not part of a qlpack", async () => {
+        await doVariantAnalysisTest({
+          queryPath: "data-remote-no-qlpack/in-pack.ql",
+          filesThatExist: ["in-pack.ql"],
+          filesThatDoNotExist: ["lib.qll", "not-in-pack.ql"],
+          qlxFilesThatExist: ["in-pack.qlx"],
+        });
+      });
+
+      it("should run a remote query that is nested inside a qlpack", async () => {
+        await doVariantAnalysisTest({
+          queryPath: "data-remote-qlpack-nested/subfolder/in-pack.ql",
+          filesThatExist: ["subfolder/in-pack.ql", "otherfolder/lib.qll"],
+          filesThatDoNotExist: ["subfolder/not-in-pack.ql"],
+          qlxFilesThatExist: ["subfolder/in-pack.qlx"],
+        });
+      });
+    });
+
+    async function doVariantAnalysisTest({
+      queryPath,
+      filesThatExist,
+      qlxFilesThatExist,
+      filesThatDoNotExist,
+    }: {
+      queryPath: string;
+      filesThatExist: string[];
+      qlxFilesThatExist: string[];
+      filesThatDoNotExist: string[];
+    }) {
+      const fileUri = getFile(queryPath);
+
+      await variantAnalysisManager.runVariantAnalysis(
```

Comment:
Since this is testing the query pack generation instead of specifically the variant analysis, would it make sense to make these tests of a new test suite for [`run-remote-query.ts`](https://github.com/github/vscode-codeql/blob/6f37f176e4a6aa474d2574ff632b3f5f2828da20/extensions/ql-vscode/src/variant-analysis/run-remote-query.ts) which tests the `prepareRemoteQueryRun` function? I think that makes the test easier since you don't need to mock anything, although you'll lose some of the end-to-end testing that we're doing now.

Reply:
I started going down this path, but setting up the parameters for `prepareRemoteQueryRun` is no simpler than using the `variantAnalysisManager` already existing. Since the tests are already set up for this, we can keep things simpler by reusing existing test infrastructure.